openedx/core/djangoapps/user_api/views.py

```diff
@@ -249,6 +251,16 @@ def get(self, request):
         form_desc = FormDescription("post", reverse("user_api_registration"))
         self._apply_third_party_auth_overrides(request, form_desc)
 
+        if configuration_helpers.get_value(
+            'ALLOW_REGISTRATION_FORM_FIELD_OVERRIDE',
+            settings.FEATURES.get('ALLOW_REGISTRATION_FORM_FIELD_OVERRIDE', False)):
+            form_desc.override_field_properties(
+                'country',
+                default=configuration_helpers.get_value(
+                    'REGISTRATION_COUNTRY_FIELD_DEFAULT',
+                    settings.FEATURES.get('REGISTRATION_COUNTRY_FIELD_DEFAULT', ''))
+            )
```

Comment:
This logic will probably have to be a bit more generic to get accepted upstream. There's no particular reason upstream would be cool with it if we just allow overriding the country field's default without considering other fields.

One way of achieving that could be to add a configuration variable named `REGISTRATION_FIELD_DEFAULTS`, which is a list of dictionaries of the form:

```yml
REGISTRATION_FIELD_DEFAULTS:
- field: country
  default: "USA"
- field: gender
  default: "Male"
...
```

And then we can simply loop through this list and apply variables as appropriate.